src/vector/geometry.rs

```diff
@@ -197,3 +217,11 @@ impl Drop for Geometry {
         }
     }
 }
+
+impl Clone for Geometry {
+    fn clone(&self) -> Geometry {
+        let c_geometry = self.c_geometry_ref.borrow();
+        let new_c_geom = unsafe { ogr::OGR_G_Clone(c_geometry.unwrap())};
```

Comment:
Should I just check `self.has_gdal_ptr()` is true before allowing to get the `c_geometry` and performing the cloning ? I guess `clone` cannot return a `Result` so i don't know the preferred way to handle errors here, i will look into it.